runtime/src/blueoil_image.cpp

```diff
@@ -0,0 +1,194 @@
+
+namespace blueoil {
+namespace image {
+
+template <typename T>
+T clamp(const T x, const T lowerLimit, const T upperLimit) {
+    if (x < lowerLimit) {
+	return lowerLimit;
+    }
+    if (upperLimit < x) {
+	return upperLimit;
+    }
+    return x;
+}
+
+/*
+ * return RGB float array
+ */
+float *Tensor_at(Tensor &tensor, const int x, const int y) {
+    const int height = tensor.shape[0];
+    const int width  = tensor.shape[1];
+    const int channels  = tensor.shape[2];
+    if ((channels != 1) && (channels != 3)) {
+	throw "wrong channles != 1,3"; // XXX
+    }
+    const int clamped_x = clamp(x, 0, width-1);
+    const int clamped_y = clamp(y, 0, height-1);
+    const int scanlineSize = width * channels;
+    float *imagePtr = &(tensor.data[0]);
+    float *scanlinePtr = imagePtr + clamped_y * scanlineSize;
+    float *pixelPtr = scanlinePtr + clamped_x * channels;
+    return pixelPtr;
+}
+
+Tensor Tensor_CHW_to_HWC(Tensor &tensor) {
+    const int channels  = tensor.shape[0];
+    const int height = tensor.shape[1];
+    const int width  = tensor.shape[2];
+    if ((channels != 1) && (channels != 3)) {
+	throw "wrong channles != 1,3"; // XXX
+    }
+    Tensor dstTensor = Tensor::zeros({height, width, channels});
+    int srcPlaneSize = width * height;
+    float *srcImagePtr = &(tensor.data[0]);
+    float *dstImagePtr = &(dstTensor.data[0]);
+    for (int y = 0 ; y < height ; y++) {
+	for (int x = 0 ; x < width ; x++) {
+	    float *srcPixelPtr0 = srcImagePtr + x + (y * height);
+	    for (int c = 0 ; c < channels ; c++) {
+		float *srcPixelPtr = srcPixelPtr0 + (c * srcPlaneSize);
+		*dstImagePtr = *srcPixelPtr;
+		dstImagePtr++;
+	    }
+	}
+    }
+    return dstTensor;
+}
+
+Tensor Tensor_HWC_to_CHW(Tensor &tensor) {
+    int height = tensor.shape[0];
+    int width  = tensor.shape[1];
+    int channels  = tensor.shape[2];
+    if ((channels != 1) && (channels != 3)) {
+	throw "wrong channles != 1,3"; // XXX
+    }
+    Tensor dstTensor = Tensor::zeros({channels, height, width});
+    float *srcImagePtr = &(tensor.data[0]);
+    float *dstImagePtr = &(dstTensor.data[0]);
+    for (int c = 0 ; c < channels ; c++) {
+	float *srcPixelPtr = srcImagePtr + c;
+	for (int y = 0 ; y < height ; y++) {
+	    for (int x = 0 ; x < width ; x++) {
+		*dstImagePtr = *srcPixelPtr;
+		srcPixelPtr += 3;
+		dstImagePtr++;
+	    }
+	}
+    }
+    return dstTensor;
+}
+
+/*
+ * Resize Image
+ */
+Tensor ResizeHorizontal(Tensor &tensor, const int width,
+			const enum ResizeFilter filter) {
+    if (filter != RESIZE_FILTER_NEAREST_NEIGHBOR) {
+	throw "unknown ResizeFilter:"; // XXX
+    }
+    const int srcHeight = tensor.shape[0];
+    const int srcWidth  = tensor.shape[1];
+    const int channels  = tensor.shape[2];
+    const int height = srcHeight;
+    Tensor dstTensor = Tensor::zeros({height, width, channels});
+    float xScale = (float) width / (float) srcWidth;
+    int xSrcWindow = floorf(1/xScale);
+    xSrcWindow = 1; // Nearest
```

Comment:
As we unconditionally assigning 1 here initializing it with `floor(1/xScale)` makes no sense...
How about just initializing the value with 1?
```cpp
int xSrcWindow = 1;
```